server/parameters/ServerSubcommandParser.java

```diff
@@ -0,0 +1,154 @@
+            if (iterate(options).noneMatch(option -> option.name().equals(required.name()))) {
+                throw TypeDBException.of(CLI_OPTION_REQUIRED, required.helpEntry(""));
+            }
+        });
+    }
+
+    private static void validateUnrecognisedArgs(Set<OptionParser> recognisedParsers, Set<Option> options) {
+        recognisedParsers.forEach(parser -> {
+            if (iterate(options).noneMatch(option -> option.name().equals(parser.name()))) {
+                throw TypeDBException.of(CLI_OPTION_UNRECOGNISED, parser);
+            }
+        });
+    }
+
+    public static class Server extends SubcommandParser<ServerSubcommand.Server> {
+
+        private static final OptionParser.Flag debugParser = new OptionParser.Flag("debug", "Run server in debug mode.");
+        private static final OptionParser.Flag helpParser = new OptionParser.Flag("help", "Print help menu.");
+        private static final OptionParser.Flag versionParser = new OptionParser.Flag("version", "Print version number.");
+        private static final OptionParser.Path configPathParser = new OptionParser.Path("config", "Path to TypeDB YAML configuration file.");
+        private static final Set<OptionParser> auxiliaryParsers = set(debugParser, helpParser, versionParser, configPathParser);
+        private static final String[] tokens = new String[]{};
+        private static final String description = "Run TypeDB server";
+
+        private final ConfigParser configParser;
+
+        public Server(ConfigParser configParser) {
+            super(tokens, description);
+            this.configParser = configParser;
+        }
+
+        @Override
+        protected ServerSubcommand.Server parse(Set<Option> options) {
+            Set<Option> auxOptions = findAuxiliaryOptions(options);
+            Set<Option> configOptions = excludeOptions(options, auxOptions);
+            Optional<Path> configPath = configPathParser.parse(auxOptions);
+            Config config = configPath
+                    .map(path -> ConfigFactory.create(path, configOptions, configParser))
+                    .orElse(ConfigFactory.create(configOptions, configParser));
+            return new ServerSubcommand.Server(debugParser.parse(auxOptions), helpParser.parse(auxOptions),
+                    versionParser.parse(auxOptions), config);
+        }
+
+        private Set<Option> findAuxiliaryOptions(Set<Option> options) {
+            return iterate(options).filter(opt1 -> iterate(auxiliaryParsers).anyMatch(opt2 -> opt2.name.equals(opt1.name()))).toSet();
+        }
+
+        private Set<Option> excludeOptions(Set<Option> options, Set<Option> exclude) {
+            return iterate(options).filter(opt -> !exclude.contains(opt)).toSet();
+        }
+
+        @Override
+        public List<HelpEntry> helpEntries() {
+            List<HelpEntry> aux = list(helpParser.helpEntry(""), versionParser.helpEntry(""), debugParser.helpEntry(""), configPathParser.helpEntry(""));
```

Comment:
got confused on first reading, I think this could be cleared up by calling this `auxEntries`, so that the line below looks like it's concatenating two lists